mime.go

```diff
@@ -107,8 +120,96 @@ func (mh *mimeHandler) initJPEGEncoder(frame *C.AVFrame) error {
 	return nil
 }
 
+func (mh *mimeHandler) convertYUYV(frame *C.AVFrame) ([]byte, camera.ImageMetadata, error) {
+	if mh.yuyvSwsCtx == nil || frame.width != mh.yuyvFrame.width || frame.height != mh.yuyvFrame.height {
+		if err := mh.initYUYVContext(frame); err != nil {
+			return nil, camera.ImageMetadata{}, err
+		}
+	}
+	if mh.yuyvSwsCtx == nil {
+		return nil, camera.ImageMetadata{}, errors.New("failed to create YUYV converter")
+	}
+	if mh.yuyvFrame == nil {
+		return nil, camera.ImageMetadata{}, errors.New("failed to create yuyv destination frame")
+	}
+	res := C.sws_scale(
+		mh.yuyvSwsCtx,
+		(**C.uint8_t)(unsafe.Pointer(&frame.data[0])),
+		(*C.int)(unsafe.Pointer(&frame.linesize[0])),
```

Comment:
does a nil check on frame exist elsewhere?

Reply:
Good call. Added nil checks to convert methods.